src/helpers/Monitor.cpp

```diff
@@ -553,7 +553,12 @@ void CMonitor::changeWorkspace(CWorkspace* const pWorkspace, bool internal) {
 
         if (const auto PLASTWINDOW = pWorkspace->getLastFocusedWindow(); PLASTWINDOW)
             g_pCompositor->focusWindow(PLASTWINDOW);
-        else {
+        else if (g_pConfigManager->getConfigValuePtr("input:follow_mouse")->intValue != 1) {
+            CWindow* pWindow = g_pCompositor->getTopLeftWindowOnWorkspace(pWorkspace->m_iID);
```

Comment:
two things:
first, why not make this default?

second, we still need a simulateMouseMovement at the end to avoid wonky cursor coords if the mouse isnt moved

Reply:
> first, why not make this default?

This is because the window under curouse would be focused if follow_mouse were set to 1.

> second, we still need a simulateMouseMovement at the end to avoid wonky cursor coords if the mouse isnt moved

Will update

Reply:
> This is because the window under cursor would be focused if follow_mouse were set to 1.

This is probably going to sound a bit silly, but should it be put to a vote?

Reply:
> This is because the window under cursor would be focused if follow_mouse were set to 1.

then focus it (if there is one) and use the rest of the logic if there isn't...?

Reply:
Sorry, I don't understand. Are you talking about the conditions where `follow_mouse == 1`?

Reply:
Well, the current main branch behaves like this when `follow_mouse == 1`.

Reply:
I will be AFK, there is another branch [fix_2451.alt](https://github.com/memchr/Hyprland/tree/fix_2451.alt) which contains an alternative where this is the default (no follow_mouse check). If this behaviour is desired as the default, you can force push this branch to the current pull request branch. 

- IIRC GitHub already allows maintainers to push to the pull request branch on the fork by default.

Thanks for your time

Reply:
check my commit out

Reply:
Splendid, that will do.